e2e/e2e_test.go

```diff
@@ -0,0 +1,363 @@
+			_, err = applicationsClient.Create(&testcase.application)
+			if err != nil {
+				t.Fatalf("failed to create application: %s", err)
+			}
+
+			defer func() {
+				err := applicationsClient.Delete(testcase.application.Name, &metav1.DeleteOptions{})
+				if err != nil && !apimachineryerrors.IsNotFound(err) {
+					t.Fatalf("failed to delete application: %s", err)
+				}
+			}()
+			actual, err := applicationsClient.Get(testcase.application.Name, metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("failed to get application: %s", err)
+			}
+
+			assert.Equal(t, expected.ObjectMeta.Name, actual.ObjectMeta.Name)
+			assert.Equal(t, expected.ObjectMeta.Labels, actual.ObjectMeta.Labels)
+			assert.Equal(t, expected.ObjectMeta.Annotations, actual.ObjectMeta.Annotations)
+			assert.Equal(t, expected.Spec, actual.Spec)
+
+		})
+	}
+}
+
+var applicationStatusTests = []struct {
+	expectedYamlFilePath string
+	applicationStatus    v1.ApplicationStatus
+}{
+	{"expected/applicationstatus/simple-success.yml", v1.ApplicationStatus{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "simple-success-aabbcc112233",
+			Namespace: "default",
+			Labels: map[string]string{
+				"app":                 "simple-success",
+				"fiaas/deployment_id": "ddeeff556677",
+			},
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "fiaas.schibsted.io/v1",
+			Kind:       "ApplicationStatus",
+		},
+		Result: "SUCCESS",
+	}},
+}
+
+func TestApplicationStatus(t *testing.T) {
+	clientset, err := createClient()
+	if err != nil {
+		t.Fatalf("failed to create client: %s", err)
+	}
+	for i, testcase := range applicationStatusTests {
+		t.Run(fmt.Sprintf("%d/%s", i, testcase.expectedYamlFilePath), func(t *testing.T) {
+
+			expected, err := applicationStatusFromYaml(testcase.expectedYamlFilePath)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			applicationStatusesClient := clientset.FiaasV1().ApplicationStatuses(
+				testcase.applicationStatus.Namespace)
+			defer func() {
+				err := applicationStatusesClient.Delete(testcase.applicationStatus.Name,
+					&metav1.DeleteOptions{})
+				if err != nil && !apimachineryerrors.IsNotFound(err) {
+					t.Fatalf("failed to delete applicationStatus: %s", err)
+				}
+			}()
+
+			_, err = applicationStatusesClient.Create(&testcase.applicationStatus)
+			if err != nil {
+				t.Fatalf("failed to create applicationStatus: %s", err)
+			}
+
+			actual, err := applicationStatusesClient.Get(testcase.applicationStatus.Name,
+				metav1.GetOptions{})
+			if err != nil {
+				t.Fatalf("failed to get applicationStatus: %s", err)
+			}
+
+			assert.Equal(t, expected.ObjectMeta.Name, actual.ObjectMeta.Name)
+			assert.Equal(t, expected.ObjectMeta.Labels, actual.ObjectMeta.Labels)
+			assert.Equal(t, expected.ObjectMeta.Annotations, actual.ObjectMeta.Annotations)
+			assert.Equal(t, expected.Result, actual.Result)
+			assert.Equal(t, expected.Logs, actual.Logs)
```

Comment:
You don't have a `logs` field in your tests-data, so it seems weird to test that it matches. Maybe just add a couple log lines to ensure that it works to read those?

Reply:
I've added logs to the applicationstatus test cases